cmd/internal/planetscale_edge_database.go

```diff
@@ -134,42 +134,44 @@ func (p PlanetScaleEdgeDatabase) ListShards(ctx context.Context, psc PlanetScale
 
 func (p PlanetScaleEdgeDatabase) Read(ctx context.Context, w io.Writer, ps PlanetScaleSource, s ConfiguredStream, tc *psdbconnect.TableCursor) (*SerializedCursor, error) {
 	var (
-		err     error
-		sc      *SerializedCursor
-		hasRows bool
+		err  error
+		sErr error
+		sc   *SerializedCursor
 	)
 
 	tabletType := psdbconnect.TabletType_primary
 	table := s.Stream
 	readDuration := 1 * time.Minute
-
+	preamble := fmt.Sprintf("[%v:%v shard : %v] ", table.Namespace, table.Name, tc.Shard)
 	for {
-		p.Logger.Log(LOGLEVEL_INFO, fmt.Sprintf("syncing rows for stream [%v] in namespace [%v] with cursor [%v]", table.Name, table.Namespace, tc))
-		p.Logger.Log(LOGLEVEL_INFO, "peeking to see if there's any new rows")
-
-		hasRows, _, _ = p.sync(ctx, tc, table, ps, tabletType, true)
-		if !hasRows {
-			p.Logger.Log(LOGLEVEL_INFO, "no new rows found, exiting")
+		p.Logger.Log(LOGLEVEL_INFO, preamble+"peeking to see if there's any new rows")
+		latestCursorPosition, _ := p.getLatestCursorPosition(ctx, tc.Shard, tc.Keyspace, table, ps, tabletType)
```

Comment:
If we ignore the error at the one call site this function is used, what's the purpose of the error?

Should either handle it or remove it from the signature.